mix.exs

```diff
@@ -50,12 +52,14 @@ defmodule Spear.MixProject do
       {:event_store_db_gpb_protobufs, "~> 2.2"},
       {:connection, "~> 1.0"},
       # optional dependencies
-      {:jason, ">= 0.0.0", optional: true},
+      {:jason, "~> 1.4"},
```

Comment:
I would like to keep this optional and the version range as wide as possible since we only use it in a few places as a default serialization format:

```suggestion
      {:jason, "~> 1.0", optional: true},
```

Dialyzer doesn't like optional dependencies if I remember correctly. In this case I'm ok with adding some ignores to the `.dialyzer_ignore.exs` for the Jason calls. Ideally you should be able to use Spear without needing Jason, for example if you want to use protobuf or `:erlang.term_to_binary/1` as a serialization format instead of JSON